### PR TITLE
Canvas sprite animations

### PR DESCRIPTION
**Pull request by bartvde**
SDK-717

### CODE REVIEW

**Comment by bartvde** [comment]
I think the functionality is all in place now, just needs the unit tests.

**Comment by bartvde** [comment]
starting to work on unit tests, it seems the determineTainting method of ol.style.IconImage is "crashing" jest

File: src/components/map.js [hunk 1]
```diff
@@ -647,20 +652,83 @@ export class Map extends React.Component {
...
+    for (i = 0, ii = layers.length; i < ii; ++i) {
+      const layer = layers[i];
+      spriteOptions[layer.id] = jsonClone(layer.metadata['bnd:animate-sprite']);
+      if (Array.isArray(layer.filter)) {
+        layer.filter = createFilter(layer.filter);
+      }
+    }
+    olLayer.setStyle((feature, resolution) => {
+      // loop over the layers to see which one matches
+      for (let i = 0, ii = layers.length; i < ii; ++i) {
```
Comment (theduckylittle) [inline src/components/map.js]: Does the linter throw up a warning for this line? `i` and `ii` are defined on line 663 and this could have side effects.

Comment (bartvde) [reply src/components/map.js]: the linter did not complain, but I will check it out

Comment (bartvde) [reply src/components/map.js]: I changed it, better to be safe than sorry ;-)

**Pull request merged**